scaleway/resource_account_project.go

```diff
@@ -0,0 +1,125 @@
+package scaleway
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	accountV2 "github.com/scaleway/scaleway-sdk-go/api/account/v2"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func resourceScalewayAccountProject() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceScalewayAccountProjectCreate,
+		ReadContext:   resourceScalewayAccountProjectRead,
+		UpdateContext: resourceScalewayAccountProjectUpdate,
+		DeleteContext: resourceScalewayAccountProjectDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The name of the project",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Description of the project",
+			},
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time of the creation of the Project",
+			},
+			"updated_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time of the last update of the Project",
+			},
+			"organization_id": organizationIDSchema(),
+		},
+	}
+}
+
+func resourceScalewayAccountProjectCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	accountAPI := accountV2API(meta)
+
+	res, err := accountAPI.CreateProject(&accountV2.CreateProjectRequest{
+		Name:        d.Get("name").(string),
```

Comment:
`name` is not optional, if we keep it as optional in terraform schema then it should be computed and it should be generated in Create.